internal/services/cognitive/cognitive_account_data_source.go

```diff
@@ -67,6 +69,8 @@ func dataSourceCognitiveAccount() *pluginsdk.Resource {
 				Sensitive: true,
 			},
 
+			"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
```

Comment:
This should be using the Computed schema for identity 
```suggestion
			"identity": commonschema.SystemAssignedUserAssignedIdentityComputed(),
```